File: c++/fattoriale.cpp
#include <iostream>
using namespace std;
int main()
{
	int a,b;
	cout<<"Inserisci il numero: ";cin>>a;
	if (a<0){
		cout<<"Impossibile calcolare il fattoriale del numero inserito";
	}
	else if (a==0){
		cout<<"Il fattoriale e': 1 ";
	}
	else  if (a>1) {
		b=a;
		while(b>1){
			b=b-1;
			a=b*a;
		}
		cout<<"Il fattoriale del numero inserito e' : "<<a;
	}
	return 0;
}

File: pascal/fattoriale.pas
program factoriale;
uses crt;
var
  a,b : Integer;
Begin
     write('Inserisci il numero ');
     readln(a);
     if a<0 then
        Begin
             write('Impossibile calcolare il fattoriale del numero inserito');
        End
     else if a=0 then
          Begin
               write('Il fattoriale e: 1');
          End
     else if a>0 then
          Begin
               b:=a;
               while b > 1 do
                     Begin
                     b := b-1;
                     a := a*b;
                     End;
               writeln('Il fattoriale del numero inserito e : ', a);
         End;
         readln;
End.
